### ISSUE TITLE
wat2wasm doesn't seem to accept utf escapes

### ISSUE THREAD

**Issue by codefromthecrypt** [WebAssembly/wabt#1767]
If my wasm includes `"☺\n"` or `"\e2\98\ba\n"`, it behaves normally. I would expect `"\u{263a}\n"` to end up with the same byte sequence, but it fails compilation like:

```bash
$ smiley.wat
smiley.wat:6:39: error: bad escape "\u"
    (data (global.get $start_string) "\u{263a}\n")
                                      ^^
smiley.wat:6:38: error: unexpected token Invalid, expected ).
    (data (global.get $start_string) "\u{263a}\n")
                                     ^^^^^^^^^^^^

$ wat2wasm --version
1.0.24
```

Here's the working thingy, and I checked the rust impl and that seems to be ok..
```wat
(module
    (import "env" "print_string" (func $print_string( param i32 )))
    (import "env" "buffer" (memory 1))
    (global $start_string (import "env" "start_string") i32)
    (global $string_len i32 (i32.const 4))
    (data (global.get $start_string) "\e2\98\ba\n")
    (func (export "smiley")
        (call $print_string (global.get $string_len))
    )
)
```

See https://www.w3.org/TR/wasm-core-1/#text-string

cc @alexcrichton as I think the rust impl is correct, just it could be easier to try the rust equiv of wat2wasm since you are doing the parsing anyway... if you think relevant I'll open an issue asking for a rust variant! https://github.com/bytecodealliance/wasm-tools

**Comment by zherczeg** [comment]
I have started working on this: #1909

**Comment by zherczeg** [comment]
I think this issue can be closed now.

**Comment by keithw** [comment]
Fixed in #1909
